LongevityWorldCup.Website/wwwroot/partials/leaderboard-content.html

```diff
@@ -2395,8 +2430,69 @@ <h3>Proofs</h3>
+                const ageInCompleteYears = parseInt(chronoAgeAtEntry);
+                for (let index = 0; index < ageInCompleteYears; index++) {
+                    allTimeAgingPaces.push(initialPace);
+                }
+
+                if (isEntryFromTheLastYear) {
+                    // the first entry is under 1 year ago, so estimate the pace for the previous year
+                    seasonAgingPaces.push(initialPace);
+                }
+
+                previousEntryPhenoAge = phenoAge - initialPace;
+                previousEntryChronoAge = chronoAgeAtEntry - 1;
+            } else {
+                const isThereAnotherEntry = Boolean(completeData[entryIndex + 1]);
+                const entryYear = entryDate.getFullYear().toString();
+                if (isThereAnotherEntry && completeData[entryIndex + 1].Date.startsWith(entryYear)) {
+                    // The next entry is from the same year, skip this so we can deal with them in a mini loop
+                } else {
+                    // This is either the only entry for a year, or the last one for that year
+                    const entriesForTheYear = completeDataByYear[entryYear];
+                    const countEntriesForTheYear = entriesForTheYear.length;
+                    const pacesForTheYear = [];
+                    entriesForTheYear.forEach((entryForTheYear, entryForTheYearIndex) => {
+                        const entryDate = new Date(entryForTheYear.Date);
+                        const chronoAgeAtEntry = window.calculateAgeAtDate(dob, entryDate);
+                        const biomarkerValues = prepareBiomarkerValues(entryForTheYear, chronoAgeAtEntry);
+                        const phenoAge = window.PhenoAge.calculatePhenoAge(biomarkerValues);
+
+                        if (previousEntryPhenoAge) {
+                            const phenoAgeDifference = phenoAge - previousEntryPhenoAge;
+                            const timePassed = chronoAgeAtEntry - previousEntryChronoAge;
+                            const timePassedProportionOfYear = 1 / timePassed;
+                            pacesForTheYear.push(timePassedProportionOfYear * phenoAgeDifference);
+                        }
+
+                        previousEntryPhenoAge = phenoAge;
+                        previousEntryChronoAge = chronoAgeAtEntry;
+                    });
+
+
+                    const averagePaceForTheYear = pacesForTheYear.reduce((a, b) => (a + b)) / pacesForTheYear.length;
+                    allTimeAgingPaces.push(averagePaceForTheYear);
+
+                    if (isEntryFromThisSeason) {
+                        seasonAgingPaces.push(averagePaceForTheYear);
+                    }
+                }
+            }
         });
 
+        window.averagePaceOfAging = allTimeAgingPaces.reduce((a, b) => (a + b)) / allTimeAgingPaces.length;
+        window.seasonPaceOfAging = seasonAgingPaces.reduce((a, b) => (a + b)) / seasonAgingPaces.length;
```

Comment:
If there is no item in `seasonAgingPaces` or `allTimeAgingPaces` it will divide by 0. Please handle it
(For `seasonAgingPaces` it is likely to happen)